src/argowrapper/engine/helpers/argo_engine_helper.py

```diff
@@ -59,12 +58,7 @@ def parse_common_details(
         )
 
     if user_name_str:
-        if user_name_str.startswith("user-"):
-            user_name_str = user_name_str[5:]
-        # argo engine encode % as -
-        user_name_str = urllib.parse.unquote(user_name_str.replace("-", "%"))
-        # put actual - back after decoding
-        user_name_str = user_name_str.replace("%", "-")
+        user_name_str = convert_username_label_to_gen3username(user_name_str)
```

Comment:
would be good to use this same `convert_username_label_to_gen3username` in https://github.com/uc-cdis/argo-wrapper/pull/134/commits/15b18656cdfb304bf5937ce65f5b627f4925e8e0#diff-c6a6579b39e448068e73736827d205c5b2b5b87829267f597e316b95f2f6a81cR85-R86